src/scheduler/Scheduler.java

```diff
@@ -1,5 +1,116 @@
+                switch (instruction) {
+                    case ATTRIBUTIONX:
+                        String xAttribute = instructionExecutor.split("=")[1];
+                        processPCB.setRegisterX(Integer.parseInt(xAttribute));
+                        break;
+                    case ATTRIBUTIONY:
+                        String yAttribute = instructionExecutor.split("=")[1];
+                        processPCB.setRegisterY(Integer.parseInt(yAttribute));
+                        break;
+                    case INTERRUPTION:
+                        logger.addMessage("IO_STARTED", process.getName());
+                        processIO = true;
+                        i++;
+                        break loop;
+                    case END:
+                        processEnd = true;
+                        i++;
+                        break loop;
+                    default:
+                        break;
+                }
+            }
+
+            instructionsRan.add(i);
+
+            logger.addMessage(i == 0 ? "INTERRUPTING_PROCESS_1" : "INTERRUPTING_PROCESS_2", process.getName(), i);
+            processPCB.increaseProcessQuantum();
+            processPCB.decreaseTwoCredits();
+
+            if (!processEnd) {
+                if (processIO) {
+                    process.setState(State.BLOCKED);
+                    processPCB.setWaitTo2();
+                    ProcessList.addBlockedProcess(processPCB);
+                } else {
+                    if (ProcessList.shouldContinue(processPCB)) {
+                        runAgain = true;
+                    } else {
+                        process.setState(State.READY);
+                        ProcessList.addReadyProcess(processPCB);
+                    }
+                }
+            } else {
+                logger.addMessage("ENDING_PROCESS", process.getName(), processPCB.getRegisterX(), processPCB.getRegisterY());
+                process.setState(State.FINISHED);
+                ProcessTable.removePCBofProcessTable(processPCB);
+            }
+        } while (runAgain);
+    }
+
+    public void run() {
+        while (ProcessTable.processTable.size() != 0) {
+            PCB processPCB;
+            if ((processPCB = ProcessList.removeNextInReadyList()) != null) {
+                if (processPCB.getCredit() == 0) {
+                    ProcessList.addReadyProcess(processPCB);
+                    ProcessList.resetReadyList();
+                } else {
+                    executeProcess(processPCB);
+                    changes++;
+                }
+            } else {
+                ProcessList.decreaseBlockedListWait();
+            }
+        }
+
+        int instructions = 0;
+        for (Integer instructionRunned : instructionsRan) {
+            instructions += instructionRunned;
+        }
+
+        logger.addMessage("AVERAGE_EXCHANGES", changes);
+        logger.addMessage("INSTRUCTION_AVERAGE", instructions / changes);
```

Comment:
The average is an int result every time (int/int is int), but it can be a double result